runtime/vam/expr/function/fields.go

```diff
@@ -47,11 +47,9 @@ func (f *Fields) Call(args ...vector.Any) vector.Any {
 			inOffs, outOffs = appendPaths(buildPath(rtyp, nil), s, inOffs, outOffs)
 		}
 		inner := vector.NewArray(f.innerTyp, inOffs, s, nil)
-		out := vector.NewArray(f.outerTyp, outOffs, inner, nil)
-		if len(errs) > 0 {
-			return vector.Combine(out, errs, vector.NewStringError(f.zctx, "missing", uint32(len(errs))))
-		}
-		return out
+		b := vector.NewCombiner(vector.NewArray(f.outerTyp, outOffs, inner, nil))
+		b.Add(errs, vector.NewStringError(f.zctx, "missing", uint32(len(errs))))
+		return b.Result()
```

Comment:
vector.Combine feels like a better API here than vector.NewCombiner since the latter allocates a vector.Error even when len(errs)==0. Why not stick with it?

Reply:
Doesn't work for multiple error values

Reply:
Then how about using an API that works for multiple error values when you need that and an API that doesn't do unnecessary allocations when you don't?